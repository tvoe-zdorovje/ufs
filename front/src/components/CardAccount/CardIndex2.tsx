import React from 'react'
import {
  Row,
  Col,
  Button
} from 'antd'
import renderPair from './RenderPair'

const styles = require('./CardAccount.less')

export interface CardIndex2Props {
  items: any[],
  addSidePanel: Function,
}

export default function CardIndex2({ items, addSidePanel }: CardIndex2Props) {
  const listItems = items.map((item, i) => {
    const pairs = [['Номер документа', item.docId],
      ['Очередь оплаты', item.priority],
      ['Сумма по документу', item.totalAmount],
      ['Оплачено', item.partAmount],
      ['Получатель', item.recipientShortName]]
    return (
      <Row className={styles.separatedRow} key={i}>
        <Col span={23}>
          {pairs.filter((pair) => pair[1]).map((pair, j) => renderPair(pair[0], pair[1], j))}
        </Col>
        <Col span={1}>
          <Button
            className={styles.largeIcon}
            onClick={() => addSidePanel(pairs, `ПП ${item.docId}`)}
            icon="menu-unfold"
          />
        </Col>
      </Row>
    )
  })

  return (
    <div>
      {listItems}
    </div>
  )
}
