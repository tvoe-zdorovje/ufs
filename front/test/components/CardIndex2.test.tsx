import React from 'react'
import {
  render,
  shallow
} from 'enzyme'
import toJson from 'enzyme-to-json'
import { Button } from 'antd'

import CardIndex2, { CardIndex2Props } from 'components/CardAccount/CardIndex2'

describe('Компонент CardIndex2', () => {

  const props: CardIndex2Props = {
    items: [{
      docId: '1',
      docType: '2',
      recipientShortName: '3',
      amount: '4',
      totalAmount: '5',
      paidPartly: false,
      comeInDate: '01.05.2017',
      priority: '1',
      paids: [{ date: '1', amount: '2' }]
    }],
    addSidePanel: (x, y) => {
    },
  }

  const renderWrapper = render(<CardIndex2 {...props} />)
  const shallowWrapper = shallow(<CardIndex2 {...props} />)

  it('выполняется рендер всех item', () => {
    expect(shallowWrapper.children()).toHaveLength(1)
  })

  it('работает кнопка', () => {
    const spy = jest.fn()
    shallowWrapper.setProps({ addSidePanel: spy })
    shallowWrapper.update()
    shallowWrapper.find(Button).at(0).simulate('click', {
      preventDefault() {
      }
    })
    expect(spy).toBeCalled()
  })

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})
