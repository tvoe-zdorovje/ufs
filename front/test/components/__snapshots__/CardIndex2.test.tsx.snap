// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент CardIndex2 совпадает со своим снимком 1`] = `
<div>
  <div
    class="ant-row separatedRow"
  >
    <div
      class="ant-col-23"
    >
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Номер документа
        </div>
        <div
          class="ant-col-16"
        >
          1
        </div>
      </div>
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Очередь оплаты
        </div>
        <div
          class="ant-col-16"
        >
          1
        </div>
      </div>
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Сумма по документу
        </div>
        <div
          class="ant-col-16"
        >
          5
        </div>
      </div>
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Получатель
        </div>
        <div
          class="ant-col-16"
        >
          3
        </div>
      </div>
    </div>
    <div
      class="ant-col-1"
    >
      <button
        class="ant-btn ant-btn-icon-only largeIcon"
        type="button"
      >
        <i
          class="anticon anticon-menu-unfold"
        />
      </button>
    </div>
  </div>
</div>
`;
